File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03807/p03807_num9_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> A(N);
    for(int i = 0; i < N; i++) {
        cin >> A[i];
    }
    
    int even_count = 0, odd_count = 0;
    for(int i = 0; i < N; i++) {
        if(A[i] % 2 == 0) {
            even_count++;
        } else {
            odd_count++;
        }
    }

    if(even_count % 2 == 0) {
        cout << "YES" << endl;
    } else if(odd_count == N) {
        if(even_count <= 1) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    } else {
        cout << "NO" << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03807/p03807_num9_parsed.py
# Program:

def can_have_one_integer(N, A):
    odd_count = sum(1 for num in A if num % 2 != 0)
  
    if odd_count % 2 != N % 2:
        return "NO"
    elif odd_count > 0 or odd_count == N:
        return "YES"
    else:
        return "NO"

N = int(input())
A = list(map(int, input().split()))

result = can_have_one_integer(N, A)
print(result)